zmon_worker_monitor/builtins/plugins/entities_wrapper.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+import requests
+import sys
+import urllib
+import urlparse
+import json
+from collections import defaultdict
+
+from zmon_worker_monitor.zmon_worker.errors import HttpError, CheckError, ConfigurationError
+from zmon_worker_monitor.zmon_worker.common.http import get_user_agent
+from requests.adapters import HTTPAdapter
+
+from zmon_worker_monitor.adapters.ifunctionfactory_plugin import IFunctionFactoryPlugin, propartial
+
+import tokens
+
+# will use OAUTH2_ACCESS_TOKEN_URL environment variable by default
+# will try to read application credentials from CREDENTIALS_DIR
+tokens.configure()
+tokens.manage('uid', ['uid'])
+tokens.start()
+
+logger = logging.getLogger('zmon-worker.entity-wrapper')
+
+
+class EntitiesWrapperFactory(IFunctionFactoryPlugin):
+    def __init__(self):
+        super(EntitiesWrapperFactory, self).__init__()
+
+
+    def configure(self, conf):
+        """
+        Called after plugin is loaded to pass the [configuration] section in their plugin info file
+        :param conf: configuration dictionary
+        """
+
+        self.service_url = conf.get('entityservice.url', conf.get('dataservice.url', 'https://localhost:443'))
+        self.oauth2 = conf.get('entityservice.oauth2', conf.get('dataservice.oauth2', False))
+
+        return
+
+
+    def create(self, factory_ctx):
+        """
+        Automatically called to create the check function's object
+        :param factory_ctx: (dict) names available for Function instantiation
+        :return: an object that implements a check function
+        """
+
+        return propartial(EntitiesWrapper, infrastructure_account=factory_ctx.get('entity', {}).get('infrastructure_account', None), service_url=self.service_url, oauth2=self.oauth2)
+
+
+class EntitiesWrapper(object):
+    def __init__(self, service_url, infrastructure_account, verify=True, oauth2=False, token=None):
+
+        if not service_url:
+            raise ConfigurationError('EntitiesWrapper improperly configured. URL is missing!')
+
+        self.__service_url = service_url
```

Comment:
`self.infrastructure_account` is not set here.

Also, `verify` and `token` are ignored/not used.